src/components/app.jsx

```diff
@@ -1,9 +1,58 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import { ReactComponent as Logo } from "../images/sun.svg";
 import "../App.css";
+import { CitySearch } from './CitySearch';
+import { WeeklyCards } from './WeeklyCards'; 
+import { UnitSwitch } from './UnitSwitch'; 
+import {fetchWeather} from '../utils';
 
 function App() {
-  return <h1>Weather</h1>;
+  const [weatherForecast, setWeatherForecast] = useState({});
+  let [unit, setUnit] = useState('imperial');
+  const [currentCity, setCurrentCity] = useState('New York');
+
+  
+  let handleSubmit = async (e) =>{
+    e.preventDefault();
+    window.location.replace(currentCity);
+    let data = await fetchWeather(currentCity, unit);
+    setWeatherForecast(data);
+  };
+
+  useEffect(()=>{
+     const fetchData = async () =>{
+        const [cityName, newUnit] = document.location.pathname.slice(1).replace(/%20/g, " ").split('/');
```

Comment:
This is an interesting approach to handling the URL. Were there any others you might have considered in the process?

Reply:
I could've passed them as query strings and used URLSearchParams to parse them.